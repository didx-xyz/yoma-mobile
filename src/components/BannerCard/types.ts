import { Colors } from '~/styles'

export interface Offset {
  x: number
  y: number
}

export interface Background {
  imageSrc?: number
  imageOffset?: Offset
  color?: Colors
}

export interface Content {
  title: string
  titleColor?: Colors
  body: string
  bodyColor?: Colors
  offset?: Offset
}
