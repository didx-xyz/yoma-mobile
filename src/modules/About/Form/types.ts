import { UserFields } from '~/modules/User/User.types'

export type FormFields = {
  [UserFields.Biography]: string
}
