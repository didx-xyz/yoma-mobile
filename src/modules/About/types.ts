import { NativeStackNavigationProp } from '@react-navigation/native-stack'

import { HomeNavigationRoutes, HomeNavigatorParamsList } from '~/modules/HomeNavigation/HomeNavigation.types'

export type AboutNavigation = NativeStackNavigationProp<HomeNavigatorParamsList, HomeNavigationRoutes.About>
