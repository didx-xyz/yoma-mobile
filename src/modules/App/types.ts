export enum AppActions {
  resetApp = '[App] Clear App Reducers',
}
