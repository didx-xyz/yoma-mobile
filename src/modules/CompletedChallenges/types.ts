import { NativeStackNavigationProp } from '@react-navigation/native-stack'

import { HomeNavigationRoutes, HomeNavigatorParamsList } from '~/modules/HomeNavigation/HomeNavigation.types'

export type CompleteChallengesNavigation = NativeStackNavigationProp<
  HomeNavigatorParamsList,
  HomeNavigationRoutes.CompletedChallenges
>
