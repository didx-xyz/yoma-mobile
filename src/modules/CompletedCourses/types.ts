import { NativeStackNavigationProp } from '@react-navigation/native-stack'

import { HomeNavigationRoutes, HomeNavigatorParamsList } from '~/modules/HomeNavigation/HomeNavigation.types'

export type CompletedCoursesNavigation = NativeStackNavigationProp<
  HomeNavigatorParamsList,
  HomeNavigationRoutes.Education
>
