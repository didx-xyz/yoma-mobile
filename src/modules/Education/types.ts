import { NativeStackNavigationProp } from '@react-navigation/native-stack'
import { DocumentPickerResponse } from 'react-native-document-picker'

import { UserCredentialTypes } from '~/api/users/users.types'
import { HomeNavigationRoutes, HomeNavigatorParamsList } from '~/modules/HomeNavigation/HomeNavigation.types'
import { NormalisedData } from '~/redux/redux.types'

export type EducationNavigation = NativeStackNavigationProp<HomeNavigatorParamsList, HomeNavigationRoutes.Education>

export type FormFields = {
  title: string
  type: UserCredentialTypes
  description: string
  organisationId: string
  countries: string[] | null
  startTime: string | Date | null
  endTime: string | Date | null
  skillNames: string[]
  certificate: DocumentPickerResponse | null
}

export interface Education {
  id: string
  title: string | null
  description: string | null
  instructions: string | null
  url: string | null
  createdAt: string
  zltoReward: number | null
  createdByAdmin: boolean
  language: string | null
  difficulty: string | null
  timeValue: number | null
  timePeriod: string | null
  startTime: string | null
  endTime: string | null
  published: boolean
  organisationId: string | null
  organisationName: string | null
  organisationLogoURL: string | null
  organisationURL: string | null
  organisationPrimaryContactName: string | null
  organisationPrimaryContactEmail: string | null
  organisationPrimaryContactPhone: string | null
  skills: string[] | null
  countries: string[] | null
  unverifiedCredentials: number | null
  approvedCredentials: number | null
  rejectedCredentials: number | null
  totalZLTORewarded: number | null
  skillsLearned: number | null
}

export type NormalisedEducation = NormalisedData<Education>

export type EducationState = NormalisedEducation
