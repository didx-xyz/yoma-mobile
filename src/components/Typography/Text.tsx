import React, { useMemo } from 'react'
import { Text as RNText, StyleSheet, TextProps, TextStyle } from 'react-native'

import { Colors, colors } from '~/styles'
import { WithChildren } from '~/types/react.types'

import styles from './Text.styles'
import { FontWeights, TextAlign } from './Text.types'
import { fontWeights } from './fontWeights.styles'

type Props = TextProps &
  WithChildren<{
    color?: Colors
    presetStyle?: TextStyle
    weight?: FontWeights
    align?: TextAlign
    style?: TextStyle
  }>

const Text = ({ presetStyle, weight, align, color, style, children, ...props }: Props) => {
  const colorStyle = useMemo(() => (color ? { color: colors[color] } : {}), [color])
  const fontWeightStyle = useMemo(() => (weight ? fontWeights[weight] : {}), [weight])
  const textAlignStyle = useMemo(() => (align ? { textAlign: align } : {}), [align])
  const textStyles = useMemo(
    () => StyleSheet.flatten([styles.base, presetStyle, colorStyle, fontWeightStyle, textAlignStyle, style]),
    [colorStyle, fontWeightStyle, presetStyle, style, textAlignStyle],
  )

  return (
    <RNText maxFontSizeMultiplier={1.3} minimumFontScale={0.8} style={textStyles} {...props} testID="text">
      {children}
    </RNText>
  )
}

export default Text
