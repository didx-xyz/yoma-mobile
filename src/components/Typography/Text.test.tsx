import { cleanup, render } from '@testing-library/react-native'
import React from 'react'
import 'react-native'

import { Colors, colors } from '../../styles'
import Text from './Text'
import { FontWeights, TextAlign } from './Text.types'

describe('Text', () => {
  afterEach(cleanup)

  it('should render correctly', () => {
    const { toJSON } = render(<Text>Tester</Text>)
    const componentAsJSON = toJSON()

    expect(componentAsJSON).toMatchSnapshot()
  })

  it('should have the text button defined', () => {
    const { getByTestId } = render(<Text>Tester</Text>)
    const foundTextElement = getByTestId('text')

    expect(foundTextElement).toBeDefined()
  })

  it('should correctly fallback to the default presets for color, textAlign and fontWeight if none is set', () => {
    const { getByTestId } = render(<Text>Tester</Text>)
    const foundTextElement = getByTestId('text')
    expect(foundTextElement.props.style.color).toBe(colors[Colors.PrimaryPurple])
    expect(foundTextElement.props.style.textAlign).toBe(TextAlign.left)
    expect(foundTextElement.props.style.fontWeight).toBe('500')
  })
  it('should correctly set the text color given with the color prop', () => {
    const { getByTestId } = render(<Text color={Colors.PrimaryYellow}>Tester</Text>)
    const foundTextElement = getByTestId('text')
    expect(foundTextElement.props.style.color).toBe(colors[Colors.PrimaryYellow])
  })

  it('should correctly set the text align given with the align prop', () => {
    const { getByTestId } = render(<Text align={TextAlign.right}>Tester</Text>)
    const foundTextElement = getByTestId('text')
    expect(foundTextElement.props.style.textAlign).toBe(TextAlign.right)
  })

  it('should correctly set the font weight with the weight prop', () => {
    const { getByTestId } = render(<Text weight={FontWeights.semiBold_600}>Tester</Text>)
    const foundTextElement = getByTestId('text')
    expect(foundTextElement.props.style.fontWeight).toBe('600')
  })

  it('should correctly set the style overrides when the style prop is set', () => {
    const { getByTestId } = render(<Text style={{ color: '#ffffff' }}>Tester</Text>)
    const foundTextElement = getByTestId('text')
    expect(foundTextElement.props.style.color).toBe('#ffffff')
  })

  it('should correctly pass a prop through to the underlying component', () => {
    const { getByTestId } = render(<Text numberOfLines={3}>Tester</Text>)
    const foundTextElement = getByTestId('text')
    expect(foundTextElement.props.numberOfLines).toBe(3)
  })
})
