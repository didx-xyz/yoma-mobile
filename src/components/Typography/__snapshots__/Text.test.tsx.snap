// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render correctly 1`] = `
<Text
  maxFontSizeMultiplier={1.3}
  minimumFontScale={0.8}
  style={
    Object {
      "color": "#41204B",
      "fontFamily": "Montserrat-Medium",
      "fontWeight": "500",
      "textAlign": "left",
    }
  }
  testID="text"
>
  Tester
</Text>
`;
